### PR TITLE
Fix flakiness in playwright tests

### PR DESCRIPTION
**Pull request by t3chguy**
Closes https://github.com/matrix-org/matrix-react-sdk/pull/12507

### CODE REVIEW

**Comment by richvdh** [comment]
For me, "stabilise" means something like "take out of beta"/"declare this to be a 'stable' feature". Have tweaked title accordingly.

File: playwright/element-web-test.ts [hunk 1]
```diff
@@ -316,7 +316,6 @@ export const expect = baseExpect.extend({
         name?: `${string}.png`,
         options?: {
             mask?: Array<Locator>;
-            omitBackground?: boolean;
```
Comment (richvdh) [inline playwright/element-web-test.ts:319]: This is an option supported by the underlying playwright function, so I'm not sure that ripping it out here is necessary. But if it doesn't work for us, I guess it makes sense.

File: playwright/pages/settings.ts [hunk 2]
```diff
@@ -25,8 +25,9 @@ export class Settings {
      * Open the top left user menu, returning a Locator to the resulting context menu.
      */
     public async openUserMenu(): Promise<Locator> {
-        await this.page.getByRole("button", { name: "User menu" }).click();
         const locator = this.page.locator(".mx_ContextualMenu");
+        if (await locator.isVisible()) return locator;
```
Comment (richvdh) [inline playwright/pages/settings.ts]: does this not risk finding some other context menu which happens to be open?

File: playwright/e2e/one-to-one-chat/one-to-one-chat.spec.ts [hunk 3]
```diff
@@ -45,6 +45,7 @@ test.describe("1:1 chat room", () => {
         await expect(
             page.getByRole("group", { name: "Rooms" }).locator(".mx_RoomTile").getByText(user2.displayName),
         ).not.toBeVisible();
+        await page.waitForTimeout(500); // avoid race condition with routing
```
Comment (richvdh) [inline playwright/e2e/one-to-one-chat/one-to-one-chat.spec.ts:48]: what's routing?

File: playwright/pages/settings.ts (hunk 2 above)
Comment (richvdh) [reply playwright/pages/settings.ts]: bleh I guess this is why you updated it in a later commit. More fool me for attempting to review commit-by-commit

File: playwright/element-web-test.ts (hunk 1 above)
Comment (t3chguy) [reply playwright/element-web-test.ts:319]: Good spot

File: playwright/e2e/one-to-one-chat/one-to-one-chat.spec.ts (hunk 3 above)
Comment (t3chguy) [reply playwright/e2e/one-to-one-chat/one-to-one-chat.spec.ts:48]: https://dev.to/marcomonsanto/routing-in-spas-173i

Comment (richvdh) [reply playwright/e2e/one-to-one-chat/one-to-one-chat.spec.ts:48]: yes, but do we have such a thing in Element?

**Review by richvdh — APPROVED** [review]

**Pull request merged**